src/segment-loader.js

```diff
@@ -118,6 +118,9 @@ export default class SegmentLoader extends videojs.EventTarget {
     // ...for determining the fetch location
     this.fetchAtBuffer_ = false;
 
+    // Temporary hack to not abort/downswitch post-seek
```

Comment:
Would this change make sense as part of a separate PR?

Reply:
Consider seeking to a time which is right before a segment ends. The segment will download and be appended to the buffer, and you'll seek within that buffer to right before its end. Without this, I imagine the rate selection logic will abort the download for the higher rate since the buffer is so low.

Reply:
Isn't that an issue that's present without the other change?

Reply:
Ok, I went back and verified the behaviour to make sure I clearly understand what each change does. In short, the GBL change makes the time spent at a lower rate smaller, and the `noAbortsUntil` removes it completely. So, even though they're disjoint, without the `noAbortsUntil` addition, there's still a drop to the lowest quality when seeking.

I'm gonna use the example above again but go a bit more into detail about what's actually happening so you can understand what's actually changing.

**Current behaviour**
(playing at high rate)
-> seek to right before segment ends
-> ABR selects the lowest rate because buffer is so low
-> player buffers a few segments (6 in my case) at lowest rate
(playback resumes after 1st segment complete)
-> player starts buffering higher rate segments
(resume normal behaviour)

**With GBL change**
(playing at high rate)
-> seek to right before segment ends
-> ABR selects the lowest rate because buffer is so low
-> player buffers 1 segment at lowest rate
(playback resumes)
-> player starts buffering higher rate segments
(resume normal behaviour)

**With `noAbortsUntil`**
(playing at high rate)
-> seek to right before segment ends
-> ABR selects current rate because of `noAbortsUntil`
-> player buffers 1 segment at current rate
(playback resumes)
-> player starts buffering higher rate segments
(player stalls because it can't download the next segment fast enough)
(playback resumes when the next segment finishes downloading)
(resume normal behaviour)

Notice the tradeoff with the `noAbortsUntil` -- that the player stalls.

I considered the changes "omnibus" because in my mind I was trying to fix the downswitch in quality when seeking. But I have no problem separating them -- I can see why the `noAbortsUntil` change would be contentious, and the behaviour is still improved with just the GBL changes.

Reply:
Thanks for all the examples. I see what you mean. It's an interesting question, and we may want more input on it. One thing we may also want to consider is we do have the `minimumTimeSaving` variable which is adjusted in the case where we're already rebuffering, which should count for the case of seeking.

Reply:
So... is that a "yes, split them up"?

Setting `minimumTimeSaving` to be a `segmentDuration` for 2 segments after a seek would be a clever way of keeping the same rate but still aborting on too-long downloads. Of course, if you did get an abort, you'd still be kicked down to the lowest rate; the playlist selection logic will _always_ choose the lowest rate since it'll have the least `rebufferingImpact`.

Reply:
Yeah, I think the stalling issue after 1 second of playback is what we should avoid.

In my mind, the perfect case would be: you seek, we load the same quality. If it's going to take a while we can still wait, up to a point (though that "point" is not well defined).

I think the "best" answer would be to change our seeking behavior such that we only resume playback after we have enough segment data to continue playback without a stall for our current throughput and rendition. If that takes (or will take) in excess of X time, then we consider downswitching.

However, that case may be tricky to implement. In the meantime, I don't think the abort early with downswitching is a bad situation, so long as we avoid playback stalls.

Reply:
Removed the `noAbortsUntil_` stuff